go/cmd/vtgateclienttest/services/errors.go

```diff
@@ -36,58 +43,218 @@ func newErrorClient(fallback vtgateservice.VTGateService) *errorClient {
 	}
 }
 
+// requestToError returns an error for the given request, by looking at the
+// request's prefix and requested error type. If the request doesn't match an
+// error request, return nil.
+func requestToError(request string) error {
+	if !strings.HasPrefix(request, ErrorPrefix) {
+		return nil
+	}
+	return trimmedRequestToError(strings.TrimPrefix(request, ErrorPrefix))
+}
+
+// requestToPartialError attempts to return a partial error for a given request.
```

Comment:
I think the two terms "error" and "partialerror" are imprecise, and suggest that the latter is a refinement of the former. It would be better to call two types of errors "adjective1error" and "adjective2error", whatever those adjectives might be.